VMBackup/main/parameterparser.py

```diff
@@ -111,10 +111,14 @@ def __init__(self, protected_settings, public_settings, backup_logger):
                 backup_logger.log("settings received " + str(self.dynamicConfigsFromCRP), True)
                 for config in self.dynamicConfigsFromCRP:
                     if CommonVariables.key in config and CommonVariables.value in config:
-                        if(config[CommonVariables.key] in self.wellKnownSettingFlags):
-                            self.wellKnownSettingFlags[config[CommonVariables.key]] = config[CommonVariables.value]
-                        else:
-                            backup_logger.log("The received " + str(config[CommonVariables.key]) + " is not an expected setting name.", True)
+                        found = False
+                        for flag in self.wellKnownSettingFlags:
+                            if((config[CommonVariables.key]).lower() == flag.lower()):
+                                self.wellKnownSettingFlags[flag] = config[CommonVariables.value]
+                                found = True
+                                break
```

Comment:
It might be more efficient to have another small Dictionary with lowercase name as key and actual key as value. Then you can simply look up if the lowercase key passed from CRP exists and if yes then you can assign the value by using the value as the key.

Reply:
Yep, repeated conversion of the keys(flag) into lower can be avoided at the cost of space.